packages/block-library/src/blocks/header/init.php

```diff
@@ -100,12 +104,28 @@ class="<?php echo esc_attr( join( ' ', $classes ) ); ?>"
 		if ( ! empty( $stickyRowBlock ) && ! $header_is_simple ) { ?>
 			<div class="site-header site-header--secondary site-header-sticky site-header--sticky">
 				<?php
-				echo render_block( $stickyRowBlock );
 
-				if ( $stickyRowBlock['attrs']['isPrimary']  !== true ) {
-					echo render_block( $primaryRowBlock );
+				// On all pages except articles,
+				// show sticky row and primary row,
+				// if primary is not sticky.
+				if ( ! is_single() ) {
+
+					echo render_block( $stickyRowBlock );
+
+					if ( $stickyRowBlock['attrs']['isPrimary']  !== true ) {
+						echo render_block( $primaryRowBlock );
+					}
 				}
-				?>
+
+				// On articles always show primary row,
+				// reading bar and reading progress.
+				if( is_single() && ! is_attachment() ) { ?>
+
+					<?php
+						echo render_block( $primaryRowBlock );
+					  	echo renderReadingBar();
```

Comment:
`render_reading_bar`, but I would change it to something like `get_reading_bar_markup` since `render` can be misleading in regards to what the function does